src/topology/algebra/continuous_monoid_hom.lean

```diff
@@ -231,16 +233,100 @@ end⟩⟩
 
 variables {A B C D E}
 
-instance [t2_space B] : t2_space (continuous_monoid_hom A B) :=
+@[to_additive] instance [t2_space B] : t2_space (continuous_monoid_hom A B) :=
 (is_embedding A B).t2_space
 
-instance : topological_group (continuous_monoid_hom A E) :=
+@[to_additive] instance : topological_group (continuous_monoid_hom A E) :=
 let hi := is_inducing A E, hc := hi.continuous in
 { continuous_mul := hi.continuous_iff.mpr (continuous_mul.comp (continuous.prod_map hc hc)),
   continuous_inv := hi.continuous_iff.mpr (continuous_inv.comp hc) }
 
+@[to_additive] lemma continuous_comp [locally_compact_space B] :
+  continuous (λ f : continuous_monoid_hom A B × continuous_monoid_hom B C, f.2.comp f.1) :=
+begin
+  rw (is_inducing A C).continuous_iff,
+  exact continuous_map.continuous_comp'.comp
+    ((is_inducing A B).prod_mk (is_inducing B C)).continuous,
+end
+
+@[to_additive] lemma continuous_comp_left (f : continuous_monoid_hom A B) :
+  continuous (λ g : continuous_monoid_hom B C, g.comp f) :=
+begin
+  rw (is_inducing A C).continuous_iff,
+  exact f.to_continuous_map.continuous_comp_left.comp (is_inducing B C).continuous,
+end
+
+@[to_additive] lemma continuous_comp_right (f : continuous_monoid_hom B C) :
+  continuous (λ g : continuous_monoid_hom A B, f.comp g) :=
+begin
+  rw (is_inducing A C).continuous_iff,
+  exact f.to_continuous_map.continuous_comp.comp (is_inducing A B).continuous,
+end
+
+variables (E)
+
+/-- `continuous_add_monoid_hom _ f` is a functor. -/
+@[to_additive "`continuous_monoid_hom _ f` is a functor."]
+def comp_left (f : continuous_monoid_hom A B) :
+  continuous_monoid_hom (continuous_monoid_hom B E) (continuous_monoid_hom A E) :=
+{ to_fun := λ g, g.comp f,
+  map_one' := rfl,
+  map_mul' := λ g h, rfl,
+  continuous_to_fun := f.continuous_comp_left }
+
+variables (A) {E}
+
+/-- `continuous_add_monoid_hom f _` is a functor. -/
+@[to_additive "`continuous_monoid_hom f _` is a functor."]
+def comp_right {B : Type*} [comm_group B] [topological_space B]
+  [topological_group B] (f : continuous_monoid_hom B E) :
+  continuous_monoid_hom (continuous_monoid_hom A B) (continuous_monoid_hom A E) :=
+{ to_fun := λ g, f.comp g,
+  map_one' := ext (λ a, map_one f),
+  map_mul' := λ g h, ext (λ a, map_mul f (g a) (h a)),
+  continuous_to_fun := f.continuous_comp_right }
+
 end continuous_monoid_hom
 
 /-- The Pontryagin dual of `A` is the group of continuous homomorphism `A → circle`. -/
 @[derive [topological_space, t2_space, comm_group, topological_group, inhabited]]
 def pontryagin_dual := continuous_monoid_hom A circle
+variables {A B C D E}
+
+noncomputable instance : continuous_monoid_hom_class (pontryagin_dual A) A circle :=
+continuous_monoid_hom.continuous_monoid_hom_class
+
+namespace continuous_monoid_hom
+
+/-- `pontryagin_dual` is a functor. -/
+noncomputable def dual (f : continuous_monoid_hom A B) :
```

Comment:
Might consider protecting this. There's nothing in the root namespace called `dual`, but it could conflict with a variety of things if multiple namespaces are open and their `dual`s aren't protected. If you're worried about length of the spelling, maybe some notation is in order.

Reply:
Good point. And length usually won't matter since there is still dot-notation.